### PR TITLE
Leland rails rules

### PR DESCRIPTION
**Pull request by enncoded**

### CODE REVIEW

**Comment by colleend** [comment]
Would you mind adding the brakeman link (such as https://github.com/presidentbeef/brakeman/blob/main/lib/brakeman/checks/check_header_dos.rb) as a `source-rule-url` in metadata?) (You can reference this here: https://github.com/returntocorp/semgrep-rules/commit/2cc7a752dcf62ab7a6bce35d3c2feff8e48ecc1a)

File: ruby/rails/security/audit/rails-check-page-caching-cve.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+rules:
```
Comment (colleend) [inline ruby/rails/security/audit/rails-check-page-caching-cve.yaml:1]: For this rule, we also want a ruby rule that catches instances of code like 
```
class BooksController < ApplicationController
caches_page :show
end
```
(where show is vulnerable).
Good work on catching the page-caching gems though 👍

File: ruby/rails/security/audit/rails-check-render-dos.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+rules:
```
Comment (colleend) [inline ruby/rails/security/audit/rails-check-render-dos.yaml]: same for this one: we would also want to catch cases of `render :text` 👍

File: ruby/rails/security/audit/rails-check-response-splitting.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+rules:
+  - id: rails-check-response-splitting
```
Comment (colleend) [inline ruby/rails/security/audit/rails-check-response-splitting.yaml:2]: nice!

File: ruby/rails/security/injection/rails-check-json-parsing-rce.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
+rules:
```
Comment (colleend) [inline ruby/rails/security/injection/rails-check-json-parsing-rce.yaml:1]: this one is only for CVE-2013-0333 and not for CVE-2013-0269 right?

**Review by colleend — CHANGES_REQUESTED** [review]
Thanks for doing this!

File: ruby/rails/security/injection/rails-check-json-parsing-rce.yaml (hunk 4 above)
Comment (enncoded) [reply ruby/rails/security/injection/rails-check-json-parsing-rce.yaml:1]: Yes, I'll be working on one for 0269 next when I have time to do more Brakeman rules!

**Review by colleend — APPROVED** [review]
added in the references to brakeman, lgtm!

**Pull request merged**